pyvista/core/grid.py

```diff
@@ -670,7 +673,12 @@ def points(self) -> NumpyArray[float]:
         y = np.insert(np.cumsum(np.full(ny, dy)), 0, 0.0) + oy
         z = np.insert(np.cumsum(np.full(nz, dz)), 0, 0.0) + oz
         xx, yy, zz = np.meshgrid(x, y, z, indexing='ij')
-        return np.c_[xx.ravel(order='F'), yy.ravel(order='F'), zz.ravel(order='F')]
+        points = np.c_[xx.ravel(order='F'), yy.ravel(order='F'), zz.ravel(order='F')]
+
+        direction = self.direction_matrix
+        if not np.array_equal(direction, np.eye(3)):
+            return pyvista.Transform().rotate(direction, point=self.origin).apply(points)
```

Comment:
Small optimization, no need to make a copy here

```suggestion
            return pyvista.Transform().rotate(direction, point=self.origin).apply(points, copy=False)
```